sycl/source/detail/scheduler/scheduler.cpp

```diff
@@ -442,8 +481,8 @@ MemObjRecord *Scheduler::getMemObjRecord(const Requirement *const Req) {
 }
 
 void Scheduler::cleanupCommands(const std::vector<Command *> &Cmds) {
-  if (Cmds.empty())
-  {
+  cleanupDeferredMemObjects(false);
```

Comment:
```suggestion
  cleanupDeferredMemObjects(/*Blocking*/false);
```

Reply:
fixed in https://github.com/intel/llvm/pull/6837/commits/d4537a3c85672ebd6fe7ccee110b98c5d3f46b5b